utils/maybe_async/src/lib.rs

```diff
@@ -1,70 +1,43 @@
-// Copyright (c) Facebook, Inc. and its affiliates.
-//
-// This source code is licensed under the MIT license found in the
-// LICENSE file in the root directory of this source tree.
-
 use proc_macro::TokenStream;
-use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
-use syn::{parse_macro_input, ImplItem, TraitItem};
-
-mod parse;
-use parse::Item;
+use syn::{parse_macro_input, Expr, ItemFn, TraitItemFn};
 
-mod visit;
-use visit::AsyncAwaitRemoval;
-
-/// maybe_async attribute macro
 #[proc_macro_attribute]
-pub fn maybe_async(_args: TokenStream, input: TokenStream) -> TokenStream {
-    let mut item = parse_macro_input!(input as Item);
-
-    let token = if cfg!(feature = "async") {
-        convert_async(&mut item)
-    } else {
-        convert_sync(&mut item)
-    };
-
-    token.into()
-}
-
-fn convert_sync(input: &mut Item) -> TokenStream2 {
-    match input {
-        Item::Impl(item) => {
-            for inner in &mut item.items {
-                if let ImplItem::Fn(ref mut method) = inner {
-                    if method.sig.asyncness.is_some() {
-                        method.sig.asyncness = None;
-                    }
-                }
-            }
-            AsyncAwaitRemoval.remove_async_await(quote!(#item))
-        },
-        Item::Trait(item) => {
-            for inner in &mut item.items {
-                if let TraitItem::Fn(ref mut method) = inner {
-                    if method.sig.asyncness.is_some() {
-                        method.sig.asyncness = None;
-                    }
-                }
+pub fn maybe_async(_attr: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
Here `#[maybe_async]` is an attribute procedural macro attribute. Hence it can take some arguments e. g.

```rust
#[derive(Serialize, Debug)]
```

here `derive` is the macro and `Serialize, Debug` are the `attributes` (`_attrs`). 

In our macro we expect no arguments hence we omit them.